docs/pages/enroll-resources/workload-identity/aws-oidc-federation.mdx

```diff
@@ -0,0 +1,431 @@
+create our bucket.
+
+You can omit this step if you wish to grant access to a different service
+within AWS.
+
+<Tabs>
+<TabItem label="Terraform">
+```hcl
+// Create an S3 bucket
+resource "aws_s3_bucket" "example" {
+  // Replace "example" with a meaningful, unique name.
+  bucket = "workload-id-demo"
+}
+```
+</TabItem>
+<TabItem label="AWS Console">
+
+1. Browse to S3
+2. Select "Create bucket"
+3. Enter a meaningful, unique name for your bucket, e.g `workload-id-demo`
+4. Leave other settings as default
+5. Click "Create bucket".
+
+</TabItem>
+</Tabs>
+
+### Configuring RBAC
+
+First, you'll create an IAM role. This role will be assumed by the workload
+after it authenticates to AWS using the JWT SVID issued by Teleport Workload
+Identity.
+
+When creating the IAM role, you'll define a trust policy that controls which
+workload identities are able to assume the role. This policy will contain
+conditions which will be evaluated against the claims within the JWT SVID
+issued by Teleport Workload Identity. In our case, the only claim we want to 
+evaluate is the `sub` claim, which will contain our workload's SPIFFE ID.
+
+With the IAM role created, you will then attach an IAM policy to it to define
+what privileges the workload will have when it assumes the role.
+
+<Tabs>
+<TabItem label="Terraform">
+
+```hcl
+// Create a role that the workload identity will assume.
+resource "aws_iam_role" "example" {
+  // Choose a unique, meaningful name that describes the role and the workload
+  // that will assume it.
+  name = "workload-id-demo"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [{
+      Effect = "Allow"
+      Principal = {
+        Federated = aws_iam_openid_connect_provider.example.arn
+      }
+      Action = "sts:AssumeRoleWithWebIdentity"
+      Condition = {
+        StringEquals = {
+          "${aws_iam_openid_connect_provider.example.url}:aud" = "sts.amazonaws.com"
+          "${aws_iam_openid_connect_provider.example.url}:sub" = "spiffe://example.teleport.sh/svc/example-service"
+        }
+      }
+    }]
+  })
+}
+
+resource "aws_iam_policy" "example" {
+  // Choose a unique, meaningful name that describes what the policy grants
+  // access to.
+  name        = "workload-id-s3-full-access"
+  path        = "/"
+
+  // This example policy grants full access to AWS S3. In production, you
+  // may wish to grant a less permissive policy.
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [{
+      Action = "s3:*"
+      Effect = "Allow"
+      Resource = [
+        aws_s3_bucket.workload_id_demo.arn,
+        "${aws_s3_bucket.workload_id_demo.arn}/*"
+      ]
+    }]
+  })
+}
+
+// Attach the policy to the role
+resource "aws_iam_role_policy_attachment" "example" {
+    role       = aws_iam_role.example.name
+    policy_arn = aws_iam_policy.example.arn
+}
+```
+
+</TabItem>
+<TabItem label="AWS Console">
```

Comment:
I would find an alternative approach to placing H3s in a `TabItem`, which can make it ungainly to switch between tabs. We could have separate `Tabs` components for creating the role and the trust policy, for example.

Reply:
922bf0318b